services/src/main/scala/cromwell/services/metadata/impl/sns/HybridSnsMetadataServiceActor.scala

```diff
@@ -0,0 +1,84 @@
+ *  this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+ *  BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ *  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ *  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ *  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ *  HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ *  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ *  IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package cromwell.services.metadata.impl.sns
+
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import com.typesafe.config.Config
+import cromwell.services.metadata.MetadataService.{PutMetadataAction, PutMetadataActionAndRespond}
+import cromwell.services.metadata.impl.MetadataServiceActor
+
+
+/**
+  * A metadata service implementation which will function as a standard metadata service but also push all metadata
+  * events to AWS SNS (Simple Notification Service). This class closely follows the pattern established in the
+  * HybridPubSubMetadataServiceActor
+  *
+  * Under the hood it maintains its own MetadataServiceActor and AwsSnsMetadataServiceActor. All messages are routed
+  * to the MetadataServiceActor. PutMetadataActions are also sent to the AwsSnsMetadataServiceActor. PutMetadataActionAndRespond
+  * messages will be sent to the SnsMetadataServiceActor as a standard PutMetadataAction, i.e. only the standard
+  * metadata service will be ACKing the request.
+  *
+  * To use this actor something similar to the following should be present in the cromwell.conf file:
+  * <pre>
+  * services {
+  *   MetadataService {
+  *     class="cromwell.services.metadata.impl.sns.HybridSnsMetadataServiceActor"
+  *     config {
+  *       aws {
+  *         application-name = "cromwell"
+  *         auths = [{
+  *           name = "default"
+  *           scheme = "default"
+  *         }]
+  *         region = "us-east-1"
+  *         topicArn = "arn:aws:sns:us-east-1:1111111111111:cromwell-metadata"
+  *       }
+  *     }
+  *   }
+  * }
+  * </pre>
+  *
+  * @see cromwell.services.metadata.impl.sns.AwsSnsMetadataServiceActor
+  */
+class HybridSnsMetadataServiceActor(serviceConfig: Config, globalConfig: Config, serviceRegistryActor: ActorRef) extends Actor with ActorLogging  {
+  val standardMetadataActor: ActorRef = context.actorOf(MetadataServiceActor.props(serviceConfig, globalConfig, serviceRegistryActor))
+  val awsSnsMetadataActor: ActorRef = context.actorOf(AwsSnsMetadataServiceActor.props(serviceConfig, globalConfig, serviceRegistryActor))
```

Comment:
Might be nicer one day to separate the "standard config" and "hybrid config" into separate config sections? Just in case they have conflicting values, or to make it easier to tell which value belongs to which?